### PR TITLE
Build under GHC 8.10.2

### PR DESCRIPTION
**Pull request by nc6**
This should hopefully remain backwards compatible with 8.6.5

### CODE REVIEW

File: semantics/executable-spec/src/Control/State/Transition/Extended.hs [hunk 1]
```diff
@@ -49,7 +49,7 @@ module Control.State.Transition.Extended
     applySTSIndifferently,
     reapplySTS,
 
-    -- * Random thing
+    -- Type Random thing
```
Comment (mrBliss) [inline semantics/executable-spec/src/Control/State/Transition/Extended.hs]: Replace all accident?

File: byron/crypto/test/cardano-crypto-test.cabal [hunk 2]
```diff
@@ -45,6 +45,11 @@ library
                        -fno-warn-missing-import-lists
                        -fno-warn-safe
                        -fno-warn-unsafe
+  if impl(ghc >=8.8)
+    ghc-options:       -fno-warn-missing-deriving-strategies
+  if impl(ghc >=8.10)
+    ghc-options:       -fno-warn-missing-safe-haskell-mode
+                       -fno-warn-prepositive-qualified-module
```
Comment (mrBliss) [inline byron/crypto/test/cardano-crypto-test.cabal]: Can we do `-Wno` here? :slightly_smiling_face:

File: byron/ledger/impl/cardano-ledger.cabal [hunk 3]
```diff
@@ -169,7 +165,11 @@ library
                        -fno-warn-missing-import-lists
                        -fno-warn-safe
                        -fno-warn-unsafe
-
+  if impl(ghc >=8.8)
+    ghc-options:       -fno-warn-missing-deriving-strategies
+  if impl(ghc >=8.10)
+    ghc-options:       -fno-warn-missing-safe-haskell-mode
+                       -fno-warn-prepositive-qualified-module
```
Comment (mrBliss) [inline byron/ledger/impl/cardano-ledger.cabal]: And here too :slightly_smiling_face:

File: cabal.project [hunk 4]
```diff
@@ -15,16 +15,16 @@ packages:
 
 
 constraints:
-  hashable < 1.3
+-- hashable < 1.3
```
Comment (mrBliss) [inline cabal.project]: Can go?

File: byron/ledger/impl/test/cardano-ledger-test.cabal [hunk 5]
```diff
@@ -125,6 +122,10 @@ library
                        -fno-warn-missing-import-lists
                        -fno-warn-safe
                        -fno-warn-unsafe
-
+  if impl(ghc >=8.8)
+    ghc-options:       -fno-warn-missing-deriving-strategies
+  if impl(ghc >=8.10)
+    ghc-options:       -fno-warn-missing-safe-haskell-mode
+                       -fno-warn-prepositive-qualified-module
```
Comment (mrBliss) [inline byron/ledger/impl/test/cardano-ledger-test.cabal]: And here too (also lines 121-124).

File: semantics/executable-spec/src/Control/State/Transition/Extended.hs (hunk 1 above)
Comment (nc6) [reply semantics/executable-spec/src/Control/State/Transition/Extended.hs]: Whoops!

File: cabal.project (hunk 4 above)
Comment (nc6) [reply cabal.project]: Yep, commented it to test

**Review by JaredCorduan — APPROVED** [review]
LGTM

**Comment by mrBliss** [comment]
> This should hopefully remain backwards compatible with 8.6.5

Without CI checking it, I doubt it :grin:.

Was there not a plan to include 8.6.5 in CI, but just for one platform?

Actually, this PR does break 8.6.5 support, see https://github.com/input-output-hk/goblins/commit/4c7bff1b8e72afb5f50c6cb1900e57369cbd4c6e#r41725426.

Moreover, as long as the PRs in -prelude and -base haven't been merged, we shouldn't merge this one.

In the meantime, I'll do the same for `ouroboros-network` :slightly_smiling_face:

**Comment by nc6** [comment]
> > This should hopefully remain backwards compatible with 8.6.5
> 
> Without CI checking it, I doubt it .
> 
> Was there not a plan to include 8.6.5 in CI, but just for one platform?
> 
> Actually, this PR does break 8.6.5 support, see [input-output-hk/goblins@4c7bff1#r41725426](https://github.com/input-output-hk/goblins/commit/4c7bff1b8e72afb5f50c6cb1900e57369cbd4c6e#r41725426).
> 
> Moreover, as long as the PRs in -prelude and -base haven't been merged, we shouldn't merge this one.
> 
> In the meantime, I'll do the same for `ouroboros-network`

Yes, I've fixed the issues that break things in 8.6.5

I think I might push a commit which reverts the CI to 8.6.5, but then merge the rest of the changes here.

**Comment by nc6** [comment]
Ah, except for the base update!

**Pull request merged**